src/api/mod.rs

```diff
@@ -964,8 +964,9 @@ impl<T: Pixel> ContextInner<T> {
       output_frameno_in_gop,
       self.gop_input_frameno_start[&output_frameno]
     );
-    let frames_needed = input_frameno + self.inter_cfg.keyframe_lookahead_distance();
-    if frames_needed > self.next_lookahead_frame && self.needs_more_frames(frames_needed) {
+
+    self.compute_scenechange_lookahead(input_frameno);
```

Comment:
I think the general idea for the MV-based scene cut detection was to:
1. Get rid of the "global mutable state" in `build_frame_properties()` (particularly `gop_input_frameno_start` and `gop_output_frameno_start`);
2. Compute frame invariants and the lookahead motion vectors, then run the scenecut detection, and then if a scenecut is detected just recompute the affected frame invariants.

I'm not sure how much moving the logic here affects the amount of refactoring that will need to be done later for 2. At the very least, I think that respecting minimum and maximum GOP sizes was a little more obvious when it was outside of `build_frame_properties`, but I might be wrong.

Reply:
The reason I moved it the way I did was to remove one place we depend on `self.next_lookahead_frame`. That piece of global mutable state was/still is a pain point. There's definitely still significant refactor needed to move to the way we want lookahead to work.

Reply:
Never mind, I have to revert this, I found a subtle bug where I had forgotten to remove one of the references to `self.next_lookahead_frame`.